storybook/pages/CurrencyAmountInputPage.qml

```diff
@@ -0,0 +1,125 @@
+    id: root
+
+    orientation: Qt.Horizontal
+
+    Item {
+        SplitView.fillWidth: true
+        SplitView.fillHeight: true
+
+        CurrencyAmountInput {
+            id: input
+            anchors.centerIn: parent
+            currencySymbol: ctrlCurrencySymbol.text
+            decimals: ctrlDecimals.value
+            readOnly: ctrlReadOnly.checked
+            enabled: ctrlEnabled.checked
+            value: ctrlAmount.text
+        }
+    }
+
+    LogsAndControlsPanel {
+        SplitView.minimumWidth: 300
+        SplitView.preferredWidth: 400
+
+        SplitView.fillHeight: true
+
+        ColumnLayout {
+            Layout.fillWidth: true
+            RowLayout {
+                Label {
+                    text: "Value:\t"
+                }
+                TextField {
+                    id: ctrlAmount
+                    text: "0.10"
+                    placeholderText: "Numeric value"
+                    onEditingFinished: input.value = text
+                }
+            }
+            RowLayout {
+                Label {
+                    text: "Currency:\t"
+                }
+                TextField {
+                    id: ctrlCurrencySymbol
+                    text: "EUR"
+                    placeholderText: "Currency symbol"
+                }
+            }
+            RowLayout {
+                Label {
+                    text: "Decimals:\t"
+                }
+                SpinBox {
+                    id: ctrlDecimals
+                    from: 0
+                    to: 18
+                    value: 2
+                }
+            }
+            Switch {
+                id: ctrlReadOnly
+                text: "Read only"
+            }
+            Switch {
+                id: ctrlEnabled
+                text: "Enabled"
+                checked: true
+            }
+            RowLayout {
+                Layout.fillWidth: true
+                Layout.topMargin: 16
+                Label {
+                    text: "Numeric value:"
+                }
+                Label {
+                    Layout.fillWidth: true
+                    Layout.alignment: Qt.AlignRight
+                    horizontalAlignment: Text.AlignRight
+                    font.bold: true
+                    text: input.valueAsString
+                }
+            }
+            RowLayout {
+                Layout.fillWidth: true
+                Label {
+                    text: "Valid:"
+                }
+                Label {
+                    Layout.fillWidth: true
+                    Layout.alignment: Qt.AlignRight
+                    horizontalAlignment: Text.AlignRight
+                    font.bold: true
+                    text: input.valid ? "true" : "false"
+                }
+            }
+            RowLayout {
+                Layout.fillWidth: true
+                Label {
+                    text: "Formatted as locale string:"
+                }
+                Label {
```

Comment:
It actually doesn't show `"Formatted as locale string:"` but just the text from the component:

![Screenshot from 2024-01-08 22-28-25](https://github.com/status-im/status-desktop/assets/20650004/3ecb84ae-b0a3-434c-9e9d-161e8ed9ff4f)

Reply:
Right... it should probably show `NaN` if the input (internal value) is invalid